protostar/commands/invoke_command.py

```diff
@@ -0,0 +1,145 @@
+from protostar.starknet_gateway import GatewayFacadeFactory, SuccessfulInvokeResponse
+from protostar.starknet_gateway.gateway_facade import Fee
+
+
+class InvokeCommand(ProtostarCommand):
+    def __init__(
+        self,
+        logger: Logger,
+        gateway_facade_factory: GatewayFacadeFactory,
+    ):
+        self._logger = logger
+        self._gateway_facade_factory = gateway_facade_factory
+
+    @property
+    def name(self) -> str:
+        return "invoke"
+
+    @property
+    def description(self) -> str:
+        return "Sends an invoke transaction V1 to the StarkNet sequencer."
+
+    @property
+    def example(self) -> Optional[str]:
+        return None
+
+    @property
+    def arguments(self):
+        return [
+            *SignableCommandUtil.signable_arguments,
+            *NetworkCommandUtil.network_arguments,
+            ProtostarArgument(
+                name="contract-address",
+                description="The address of the contract being called.",
+                type="int",
+                is_required=True,
+            ),
+            ProtostarArgument(
+                name="function",
+                description="The name of the function being called.",
+                type="str",
+                is_required=True,
+            ),
+            ProtostarArgument(
+                name="inputs",
+                description="The name of the function being called.",
+                type="felt",
+                is_required=True,
+                is_array=True,
+            ),
+            ProtostarArgument(
+                name="max-fee",
+                description=(
+                    "The maximum fee that the sender is willing to pay for the transaction. "
+                    'Provide "auto" to auto estimate the fee.'
+                ),
+                type="fee",
+            ),
+            ProtostarArgument(
+                name="wait-for-acceptance",
+                description="Waits for transaction to be accepted on chain.",
+                type="bool",
+                default=False,
+            ),
+        ]
+
+    async def run(self, args: Any):
+        network_command_util = NetworkCommandUtil(args, self._logger)
+        signable_command_util = SignableCommandUtil(args, self._logger)
+        network_config = network_command_util.get_network_config()
+        gateway_client = network_command_util.get_gateway_client()
+        signer = signable_command_util.get_signer(network_config)
+
+        return await self.invoke(
+            address=args.contract_address,
+            function=args.function,
+            inputs=args.inputs,
+            gateway_client=gateway_client,
+            signer=signer,
+            max_fee=args.max_fee,
+            wait_for_acceptance=args.wait_for_acceptance,
+            account_address=args.account_address,
+        )
+
+    async def invoke(
+        self,
+        address: int,
+        function: str,
```

Comment:
[nitpick]: those names are ambiguous - `contract_address` and `function_name` are a bit longer but do clarify code significantly

Reply:
I tried to be consistent with `starknet invoke`. I'll make names more explicit.

https://docs.starknet.io/documentation/develop/CLI/commands/#starknet_invoke

Reply:
Renamed. Resolving.